src/lightning/fabric/utilities/cloud_io.py

```diff
@@ -70,3 +75,31 @@ def _atomic_save(checkpoint: Dict[str, Any], filepath: Union[str, Path]) -> None
     torch.save(checkpoint, bytesbuffer)
     with fsspec.open(filepath, "wb") as f:
         f.write(bytesbuffer.getvalue())
+
+
+def is_dir(fs: AbstractFileSystem, path: Union[str, Path], strict: bool = False) -> bool:
```

Comment:
Could we make this a protected function? 
```suggestion
def _is_dir(fs: AbstractFileSystem, path: Union[str, Path], strict: bool = False) -> bool:
```

Reply:
done